pkg/helm/helm.go

```diff
@@ -112,27 +113,58 @@ func InstallOrUpgradeHelmChartFromPlugin(ctx context.Context, local client.Clien
 	return nil
 }
 
-// ChartTest to do helm test on the plugin
-func ChartTest(restClientGetter genericclioptions.RESTClientGetter, plugin *greenhousev1alpha1.Plugin) (bool, error) {
-	var hasTestHook bool
+// ChartTest executes Helm chart tests and logs test pod logs if a test fails.
+func ChartTest(ctx context.Context, restClientGetter genericclioptions.RESTClientGetter, plugin *greenhousev1alpha1.Plugin) (hasTestHook bool, testPodLogs string, err error) {
 	cfg, err := newHelmAction(restClientGetter, plugin.Spec.ReleaseNamespace)
 	if err != nil {
-		return hasTestHook, err
+		return hasTestHook, "", err
 	}
-	rta := action.NewReleaseTesting(cfg)
-	rta.Timeout = GetHelmTimeout() // set a timeout for the release testing to avoid waiting forever
-	results, err := rta.Run(plugin.Name)
+
+	testAction := action.NewReleaseTesting(cfg)
+	testAction.Timeout = GetHelmTimeout() // set a timeout for the release testing to avoid waiting forever
+	// Used for fetching logs from test pods
+	testAction.Namespace = plugin.Spec.ReleaseNamespace
+	results, err := testAction.Run(plugin.Name)
 	if err != nil {
-		return hasTestHook, err
+		release, getErr := getLatestRelease(cfg, plugin.Name)
+		if getErr != nil {
+			log.FromContext(ctx).Error(getErr, "Failed to get latest release", "plugin", plugin.Name)
+		} else {
+			var err2 error
+			testPodLogs, err2 = printTestPodLogs(ctx, testAction, release)
+			if err2 != nil {
+				log.FromContext(ctx).Error(err2, "Failed to retrieve test pod logs", "plugin", plugin.Name)
+			}
+		}
+		return hasTestHook, testPodLogs, err
```

Comment:
With an early return on error we can remove one level of indentation. The `err` returned from the first function will create a new local variable shadowing the outer scoped error. So we do not overwrite the outer error.

```suggestion
		release, err := getLatestRelease(cfg, plugin.Name)
		if err != nil {
			log.FromContext(ctx).Error(getErr, "Failed to get latest release", "plugin", plugin.Name)
			return hasTestHook, "", err
		}
		testPodLogs, err = printTestPodLogs(ctx, testAction, release)
		if err != nil {
			log.FromContext(ctx).Error(err, "Failed to retrieve test pod logs", "plugin", plugin.Name)
		}
		return hasTestHook, testPodLogs, err
```

Reply:
Thank you very much for the suggestion. Fixed in bffcd5e76e2a5da8004984a7bebd767c68546091